app/services/account_reset/pending_request_for_user.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+module AccountReset
+  class PendingRequestForUser
+    include AccountResetConcern
+    attr_reader :user
+
+    def initialize(user)
+      @user = user
+    end
+
+    def get_account_reset_request
+      AccountResetRequest.where(
+        user: user,
+        granted_at: nil,
+        cancelled_at: nil,
+      ).where(
+        'requested_at > ?',
+        account_reset_wait_period_days(user).ago,
+      ).order(requested_at: :asc).first
+    end
+
+    def cancel_account_reset_request!(account_reset_request_id:, cancelled_at:)
+      # rubocop:disable Rails/SkipsModelValidations
+      result = AccountResetRequest.where(
+        id: account_reset_request_id,
+        user: user,
+        granted_at: nil,
+        cancelled_at: nil,
+      ).where(
+        'requested_at > ?',
+        account_reset_wait_period_days(user).ago,
+      ).update_all(cancelled_at: cancelled_at, updated_at: Time.zone.now)
+      # rubocop:enable Rails/SkipsModelValidations
+
+      if result == 1
+        NotifyUserOfRequestCancellation.new(user).call
+        true
+      else
+        false
+      end
```

Comment:
Does it matter to have a return value here? Doesn't look like we're using it.

```suggestion
      NotifyUserOfRequestCancellation.new(user).call if result == 1
```

Reply:
No, this was kind of a leftover from some of the refactoring. Slight conflict with the changes from the suggestion [here](https://github.com/18F/identity-idp/pull/11329#discussion_r1793829126) so I've added 30e3065eff03fd5ca2016ffd69fc22cccc1bdc96 to incorporate the suggestion.